### ISSUE TITLE
[object_detection] Passing batch of images to inference of Mask R-CNN model won't work

### ISSUE THREAD

**Issue by Thommy257** [tensorflow/models#4720]
### System information
- **What is the top-level directory of the model you are using**: models/researc/object_detection
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.7.0
- **Bazel version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Problem
I'm crating masks to detect exact locations of cars in images, thus I'm using the pretrained COCO Models of your model zoo in combination with the object_detection_tutorial jupyter notebook. To speed up performance, I'm trying to pass a batch of images to the sess.run() command (I've modified the jupyter notebook slightly). Batch shape is (batch_size, image_width, image_height, 3) and while using Faster R-CNN models it works perfectly like this:
```
output_dict = sess.run(tensor_dict, feed_dict={image_tensor: image_batch})
```

But for some reason this won't work with Mask R-CNN models. The sess.run() command will return this error message:
```
InvalidArgumentError (see above for traceback): Tried to explicitly squeeze dimension 0 but dimension was not 1: 2
     [[Node: Squeeze_5 = Squeeze[T=DT_FLOAT, squeeze_dims=[0], _device="/job:localhost/replica:0/task:0/device:CPU:0"](detection_boxes)]
```

It seems to squeeze the image_batch back to dimension (image_width, image_height, 3), which will only work with a batch_size of 1. But why is that? Is there some explanation for this behavior or is this a bug? Is batching for the Mask R-CNN models not available? Although comments in these lines of code indicate, that it is:
```
if 'detection_masks' in tensor_dict:
            # The following processing is only for single image
            detection_boxes = tf.squeeze(tensor_dict['detection_boxes'], [0])
            detection_masks = tf.squeeze(tensor_dict['detection_masks'], [0])
            # # Reframe is required to translate mask from box coordinates to image coordinates and fit the image size.
            real_num_detection = tf.cast(tensor_dict['num_detections'][0], tf.int32)
            detection_boxes = tf.slice(detection_boxes, [0, 0], [real_num_detection, -1])
            detection_masks = tf.slice(detection_masks, [0, 0, 0], [real_num_detection, -1, -1])
            detection_masks_reframed = utils_ops.reframe_box_masks_to_image_masks(
                detection_masks, detection_boxes, image_batch.shape[1], image_batch.shape[2])
            detection_masks_reframed = tf.cast(
                tf.greater(detection_masks_reframed, 0.5), tf.uint8)
            # Follow the convention by adding back the batch dimension
            tensor_dict['detection_masks'] = tf.expand_dims(
                detection_masks_reframed, 0)
```

Clarification would help a lot! Thank you in advance!

**Comment by pkulzc** [comment]
The notebook is called "Object Detection Demo" so it's only for demo purpose. 
The function name is "run_inference_for_single_image" so it assumes batch size is 1.

To do a batched regular train, you should use model_main.py. Please see updated g3docs.

**Comment by vvsotnikov** [comment]
Hi @Thommy257, did you managed with this issue? I've encountered the same problem when tried to write some custom inference code.
